platforms/software/testing-base-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestEventSerializer.java

```diff
@@ -109,16 +112,28 @@ public void write(Encoder encoder, CompositeIdGenerator.CompositeId value) throw
         }
     }
 
-    private static class DefaultTestClassRunInfoSerializer implements Serializer<DefaultTestClassRunInfo> {
+    private static class ClassTestDefinitionSerializer implements Serializer<ClassTestDefinition> {
         @Override
-        public DefaultTestClassRunInfo read(Decoder decoder) throws Exception {
-            String className = decoder.readString();
-            return new DefaultTestClassRunInfo(className);
+        public ClassTestDefinition read(Decoder decoder) throws Exception {
+            return new ClassTestDefinition(decoder.readString());
+        }
+
+        @Override
+        public void write(Encoder encoder, ClassTestDefinition value) throws Exception {
+            encoder.writeString(value.getId());
+        }
+    }
+
+    @NullMarked
+    private static class DirectoryBasedTestDefinitionSerializer implements Serializer<DirectoryBasedTestDefinition> {
+        @Override
+        public DirectoryBasedTestDefinition read(Decoder decoder) throws Exception {
+            return new DirectoryBasedTestDefinition(new File(decoder.readString()));
         }
 
         @Override
-        public void write(Encoder encoder, DefaultTestClassRunInfo value) throws Exception {
-            encoder.writeString(value.getTestClassName());
+        public void write(Encoder encoder, DirectoryBasedTestDefinition value) throws Exception {
+            encoder.writeString(value.getTestDefintionFile().getAbsolutePath());
```

Comment:
🤔 This makes the resulting definition different after serialization. I think it might be better to call `getAbsoluteFile()` in the `DirectoryBasedTestDefinition` constructor.